ingest/api/dssapi.py

```diff
@@ -39,13 +40,20 @@ def __init__(self, url=None):
 
         self.dss_client = None
 
+        self.token = None
         self.init_dss_client()
 
-    def init_dss_client(self):
-        self.dss_client = hca.dss.DSSClient(
-            swagger_url=f'{self.url}/v1/swagger.json')
-        self.dss_client.host = self.url + "/v1"
-        self.creator_uid = 8008
+    # TODO This is workaround to not let the dss client token expired
+    # See ticket https://app.zenhub.com/workspaces/dcp-5ac7bcf9465cb172b77760d9/issues/humancellatlas/data-store/2216
+    # Create a dummy token at the same time DSS client is initialised.
+    # This is just a way to keep track when DSS client token will be expired
+    def init_dss_client(self, duration=3600 * 1000, refresh_period=60 * 10 * 1000):
+        if not self.token or self.token.is_expired():
+            self.token = Token('dummy', duration, refresh_period)
```

Comment:
Question: it looks like `self.token` is written to here but in the entire class, `self.token` isn't read. What's the purpose of assigning it here if you simply re-instantiate DSSClient before each put/get?

Reply:
Hi @maniarathi , I'm trying to utilise here the token.is_expired() as a signal to know when to reinstantiate.  I'm creating here a dummy token which will last for 1 hour (actually just 50min, because of 10min refresh period) When that dummy token is expired, a new dummy token is created at the same time DSS client is created.